app/interrupt.c

```diff
@@ -99,27 +98,47 @@ struct trap_regs {
     reg_t   dx;
     reg_t   di;
     reg_t   si;
-#ifdef __x86_64__
     reg_t   r8;
     reg_t   r9;
     reg_t   r10;
     reg_t   r11;
+    reg_t   xmm[16*2];
```

Comment:
For some reasons, I can't define a proper 8-byte wide (128b) register using the previous typedef. 

Using `__m128   xmm[16];` make it pop a Div by 0 Int immediately.

Any idea?

Reply:
Do you mean that the *compiler* crashes with an Internal Compiler Error because of a divide by zero ?

Reply:
No, compile is fine without any error.
Memtest86+ crashes at init with a Div By 0 interrupt. CS:IP all 00s.

Reply:
What if you use `movdqu` instead of `movdqa` ? Just in order to eliminate misalignment, which would normally produce #GP, as a cause of issues.

Reply:
Same. :-/

![image](https://github.com/user-attachments/assets/9ec86534-0be4-4a79-9025-662293b6999f)

Reply:
Seems to be an alignment issue, but I haven't found a better solution than manually aligning the struct by adding the r12 register. Every other GCC syntax I tried has failed.